src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -81,21 +83,50 @@ public static FunctionPointerMethodSymbol CreateMethodFromSource(FunctionPointer
                 callingConvention,
                 refKind,
                 returnType,
+                refReadonlyModifiers,
                 syntax,
                 typeBinder,
                 diagnostics,
                 suppressUseSiteDiagnostics);
         }
 
+        public static FunctionPointerMethodSymbol CreateMethodFromMetadata(CallingConvention callingConvention, ImmutableArray<ParamInfo<TypeSymbol>> retAndParamTypes)
+        {
+            Debug.Assert(retAndParamTypes.Length > 0);
+
+            ParamInfo<TypeSymbol> retInfo = retAndParamTypes[0];
+            var returnType = TypeWithAnnotations.Create(retInfo.Type, customModifiers: CSharpCustomModifier.Convert(retInfo.CustomModifiers));
+            var refCustomMods = CSharpCustomModifier.Convert(retInfo.RefCustomModifiers);
+
+            RefKind refKind = retInfo.IsByRef switch
+            {
+                false => RefKind.None,
+                true when HasInModifier(refCustomMods) => RefKind.RefReadOnly,
+                true => RefKind.Ref,
+            };
+
+            return new FunctionPointerMethodSymbol(
+                callingConvention,
+                refKind,
+                refCustomMods,
+                returnType,
+                retAndParamTypes.AsSpan()[1..]);
+        }
+
+        private static bool HasInModifier(ImmutableArray<CustomModifier> refCustomModifiers)
```

Comment:
>HasInModifier [](start = 28, length = 13)

Is this distinct from `CustomModifierUtils.HasInAttributeModifier()`?

Reply:
Only in that I did not realize the latter existed :)

---
In reply to: [374321386](https://github.com/dotnet/roslyn/pull/41340#discussion_r374321386) [](ancestors = 374321386)